inc/class-builder-post-meta.php

```diff
@@ -75,32 +76,19 @@ public function output( $post ) {
 
 	public function save_post( $post_id ) {
 
-		if ( ! $this->is_allowed_for_screen() ) {
-			return;
-		}
-
-		$nonce = null;
-		$data  = null;
-
-		if ( isset( $_POST[ $this->id . '-nonce' ] ) ) {
-			$nonce = sanitize_text_field( $_POST[ $this->id . '-nonce' ] ); // Input var okay.
+		if ( $data = $this->get_post_data() ) {
+			$this->save_data( $post_id, $data );
 		}
 
-		if ( isset( $_POST[ $this->id . '-data' ] ) ) {
-			$json = $_POST[ $this->id . '-data' ]; // Input var okay.
-			$data = json_decode( $json );
+	}
 
-			/**
-			 * Data is sometimes already slahed, see https://core.trac.wordpress.org/ticket/35408
-			 */
-			if ( json_last_error() ) {
-				$data = json_decode( stripslashes( $json ) );
-			}
+	public function wp_insert_post_data( $post_data ) {
 
-			if ( ! json_last_error()  && $nonce && wp_verify_nonce( $nonce, $this->id ) ) {
-				$this->save_data( $post_id, $data );
-			}
+		if ( $data = $this->get_post_data() ) {
+			$post_data['post_content'] = $this->get_rendered_data( $data );
```

Comment:
Data passed to/expected following `wp_insert_post_data` is:
* `sanitize_post_field` has run with the `db` context
* slashed 
* the `post_content` columns character set has been tested  and `wp_encode_emoji` run if needs be.

Some of these will be more or less relevant for our purposes here.